internal/common/helpers.go

```diff
@@ -58,3 +60,13 @@ func ToStringSlice(val interface{}) ([]string, error) {
 func CreateTestRunID() string {
 	return fmt.Sprintf("%d", rand.Intn(testRunMaxID-testRunMinID)+testRunMinID)
 }
+
+func ProcessResourceApplyResults(results resource.ApplyResults) string {
+	var errors []string
+	for _, result := range results {
+		if err := result.Err(); err != nil {
+			errors = append(errors, err.Error())
+		}
+	}
+	return strings.Join(errors, ", ")
+}
```

Comment:
Moved to the helpers since it is going to be used in both agentdeployer and servicedeployer packages.